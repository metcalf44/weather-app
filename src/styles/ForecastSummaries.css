
*, *::before, *::after {
    box-sizing: border-box;
    --btn: rgb(171, 215, 255);
    --hover: rgb(126, 193, 255);
} 

.weather-app {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    height: 85vh;
    /* max-width: 1100px; */
}

.location-details {
    font-size: 2rem;
}

.search-form {
    margin-left: 1.2em;
    width: fit-content;
    border-radius: 10px;
    box-shadow: 5px 6px 5px 0px rgba(117,117,117,0.57);
}

input {
    font-size: 1.25rem;
    padding: .25em .5em;
    border-radius: 10px 0 0 10px;
}

.search-form button {
    font-size: 1.25rem;
    padding: .25em .5em;
    border-radius: 0 10px 10px 0;
    background: var(--btn);
}

.forecast-summaries {
    display: flex;
    justify-content: space-between;
    margin: 2rem 0;
    width: 100%;
    max-width: 1000px;
}

.forecastSummary {
    padding: 1em 0;
    width: 18%;
    text-align: center;
    height: fit-content;
    border: 1px rgb(0, 0, 0) solid;
    border-radius: 10px;
    box-shadow: 7px 6px 5px 0px rgba(117,117,117,0.57);
}

.forecast-summary__date {
    font-weight: bold;
    font-size: 1.25rem;
    padding: .25em 0;
    height: fit-content;
}

.forecast-summary__icon {
    font-size: 1.75rem;
    padding: .25em 0;
}

.forecast-summary__temperature {
    font-size: 1.5rem;
}

.forecast-summary__description {
    font-size: 1.25rem;
    padding: .25em 0;
}

.forecastSummary button {
    font-size: 1rem;
    padding: .5em 1em;
    margin-top: .5em;
    border-radius: 25px;
    background: var(--btn)
}

button:hover {
    transform: scale(1.1);
    background: var(--hover);
}

.forecast-details {
    align-items: center;
    font-size: 1.25rem;
    text-align: center;
}

.forecast-details__date {
    font-weight: bold;
}

.forecast-details__date,
.forecast-details__temperature,
.forecast-details__humidity,
.forecast-details__wind-speed {
    font-size: 1.25rem;
    padding-bottom: .25em;
}

