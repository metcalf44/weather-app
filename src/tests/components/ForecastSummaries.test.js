import React from "react";
import { render } from "@testing-library/react";
import ForecastSummaries from "../../components/ForecastSummaries";

describe("ForecastSummaries", () => {
  const validProps = [
    {
      date: 1111111,
      description: "stub description",
      icon: "stubIcon",
      temperature: {
        min: 12,
        max: 22,
      },
    },
    {
      date: 2222222,
      description: "stub description",
      icon: "stubIcon2",
      temperature: {
        min: 13,
        max: 24,
      },
    },
  ];

  it("renders correctly", () => {
    const { asFragment } = render(<ForecastSummaries forecasts={validProps} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders the number of forecastSummary instances", () => {
    const { getAllByTestId } = render(
      <ForecastSummaries forecasts={validProps} />
    );

    expect(getAllByTestId("forecast-summary")).toHaveLength(2);
  });
});
