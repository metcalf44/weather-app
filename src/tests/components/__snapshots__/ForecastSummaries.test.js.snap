// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastSummaries renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-summaries"
  >
    <div
      class="forecastSummary"
      data-testid="forecast-summary"
    >
      <div
        class="forecast-summary__date"
      >
        Thu 1st Jan
      </div>
      <div
        class="forecast-summary__icon"
        data-testid="forecast-icon"
      >
        <i
          class="wi wi-day-sunny"
          name="owm"
        />
      </div>
      <div
        class="forecast-summary__temperature"
      >
        22°C
      </div>
      <div
        class="forecast-summary__description"
      >
        stub description 1
      </div>
      <button
        type="button"
      >
        More details
      </button>
    </div>
    <div
      class="forecastSummary"
      data-testid="forecast-summary"
    >
      <div
        class="forecast-summary__date"
      >
        Thu 1st Jan
      </div>
      <div
        class="forecast-summary__icon"
        data-testid="forecast-icon"
      >
        <i
          class="wi wi-day-thunderstorm"
          name="owm"
        />
      </div>
      <div
        class="forecast-summary__temperature"
      >
        24°C
      </div>
      <div
        class="forecast-summary__description"
      >
        stub description 2
      </div>
      <button
        type="button"
      >
        More details
      </button>
    </div>
  </div>
</DocumentFragment>
`;
